src/server/db_slice.cc

```diff
@@ -400,15 +400,28 @@ DbSlice::AutoUpdater::AutoUpdater(const Fields& fields) : fields_(fields) {
 
 DbSlice::AddOrFindResult& DbSlice::AddOrFindResult::operator=(ItAndUpdater&& o) {
   it = o.it;
-  exp_it = ExpireIterator{};  // ItAndUpdater doesn't have exp_it
+  exp_it = o.exp_it;
   is_new = false;
   post_updater = std::move(o).post_updater;
   return *this;
 }
 
+DbSlice::ItAndUpdater DbSlice::FindMutable(const Context& cntx, string_view key) {
+  auto [it, exp_it] = FindInternal(cntx, key, FindInternalMode::kUpdateCacheStats);
+
+  if (IsValid(it)) {
+    PreUpdate(cntx.db_index, it);
+    return {it, exp_it,
+            AutoUpdater({AutoUpdater::DestructorAction::kRun, this, cntx.db_index, it, key, true})};
+  } else {
+    return {it, exp_it, {}};
+  }
+}
+
 OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutable(const Context& cntx, string_view key,
                                                      unsigned req_obj_type) {
-  auto it = FindInternal(cntx, key, FindInternalMode::kDontUpdateCacheStats).first;
+  // Don't use FindMutable() so that we don't call PreUpdate()
+  auto [it, exp_it] = FindInternal(cntx, key, FindInternalMode::kUpdateCacheStats);
```

Comment:
why did you change to kUpdateCacheStats?

Reply:
Sorry, my bad. It was before I moved the implementation of read-only `GET` to use `FindReadOnly()`. Reverted.